packages/protocol/congestioncontrol/icca/scheduler/scheduler.go

```diff
@@ -467,8 +488,13 @@ func (s *Scheduler) schedule() *Block {
 	// remove the block from the buffer and adjust issuer's deficit
 	block := s.buffer.PopFront()
 	issuerID := identity.NewID(block.IssuerPublicKey())
-	s.updateDeficit(issuerID, new(big.Rat).SetInt64(-int64(block.Size())))
-
+	deficit := s.Deficit(issuerID)
+	if deficit.Cmp(new(big.Rat).SetInt64(int64(block.Work()))) < 0 {
+		deficitFloat, _ := deficit.Float64()
+		errorString := fmt.Sprintf("scheduler: deficit is less than block work - Deficit is %d, block work is %d, ", int(deficitFloat), block.Work())
+		panic(errorString)
+	}
```

Comment:
This check is already covered in `updateDeficit` function. 

```suggestion
```

Reply:
True, this extra check was left over from debugging.